### PR TITLE
Port enable code coverage command

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Testing : Unit testing.
PS : Manual testing will be done with task side changes.

### CODE REVIEW

File: src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs [hunk 1]
```diff
@@ -46,6 +51,80 @@ public string CommandArea
...
+                context.Output(StringUtil.Loc("IsMultiModuleParameterNotAvailable"));
+            }
+
+            string reportDirectory;
+            eventProperties.TryGetValue(EnableCodeCoverageEventProperties.ReportDirectory, out reportDirectory);
+
+            string include, exclude;
+            CodeCoverageUtilities.GetFilters(classFilter, out include, out exclude);
+
+            var codeCoverageInputs = new CodeCoverageEnablerInputs(buildFile, classFilesDirectories, include, exclude, sourceDirectories, summaryFile, reportDirectory, cCReportTask, reportBuildFile, isMultiModule);
```
Comment (nigurr) [inline src/Agent.Worker/CodeCoverage/CodeCoverageCommands.cs]: As discussed offline, take your design decision :)

File: src/Agent.Worker/CodeCoverage/CodeCoverageEnablerInputs.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        public void VerifyInputsForJacocoAnt(IExecutionContext context)
+        {
+            ClassFilesDirectories = CodeCoverageUtilities.ThrowIfParameterEmpty(ClassFilesDirectories, "ClassFilesDirectories");
+            SummaryFile = CodeCoverageUtilities.ThrowIfParameterEmpty(SummaryFile, "SummaryFile");
+            ReportDirectory = CodeCoverageUtilities.ThrowIfParameterEmpty(ReportDirectory, "ReportDirectory");
+            CCReportTask = CodeCoverageUtilities.ThrowIfParameterEmpty(CCReportTask, "CodeCoverageReportTarget");
+            ReportBuildFile = CodeCoverageUtilities.ThrowIfParameterEmpty(ReportBuildFile, "CodeCoverageReportBuildFile");
+
+            CodeCoverageUtilities.ThrowIfClassFilesDirectoriesIsInvalid(ClassFilesDirectories);
```
Comment (nigurr) [inline src/Agent.Worker/CodeCoverage/CodeCoverageEnablerInputs.cs]: Empty lines

File: src/Agent.Worker/CodeCoverage/CodeCoverageUtilities.cs [hunk 3]
```diff
@@ -110,6 +107,193 @@ public static int GetPriorityOrder(string coverageUnit)
             return (int)Priority.Other;
         }
 
+        public static string ThrowIfParameterEmpty(string parameterValue, string parameterName)
+        {
```
Comment (nigurr) [inline src/Agent.Worker/CodeCoverage/CodeCoverageUtilities.cs]: Method name and functionality are not correct.

File: src/Agent.Worker/CodeCoverage/ICodeCoverageEnabler.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        /// <summary>
+        /// Enables code coverage by editing build file
+        /// </summary>
+        void EnableCodeCoverage(IExecutionContext context, CodeCoverageEnablerInputs ccInputs);
+
+        /// <summary>
+        /// Enabler name. Name convention: CCTool_BuildTool.
+        /// </summary>
+        string Name { get; }
+    }
```
Comment (nigurr) [inline src/Agent.Worker/CodeCoverage/ICodeCoverageEnabler.cs]: better naming?

File: src/Agent.Worker/CodeCoverage/CodeCoverageUtilities.cs (hunk 3 above)
Comment (AshwiniChalla) [inline src/Agent.Worker/CodeCoverage/CodeCoverageUtilities.cs]: ThroIfParameterEmptyandTrim ?

File: src/Test/L0/Worker/CodeCoverage/CodeCoverageConstants.cs [hunk 5]
```diff
@@ -2,6 +2,7 @@
 {
     public class CodeCoverageConstants
```
Comment (nigurr) [inline src/Test/L0/Worker/CodeCoverage/CodeCoverageConstants.cs]: CodeCoverageTestConstants

File: src/Agent.Worker/CodeCoverage/CodeCoverageUtilities.cs (hunk 3 above)
Comment (nigurr) [inline src/Agent.Worker/CodeCoverage/CodeCoverageUtilities.cs]: or "TrimNonEmptyParam". If it's being called with Empty param, it's meant to throw error

**Pull request merged**